marconi-sidechain/test/Spec/Marconi/Sidechain/Integration.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Spec.Marconi.Sidechain.Integration (tests) where
+
+import Control.Lens ((<&>))
+import Control.Monad (unless)
+import Control.Monad.Catch (MonadCatch)
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad.Trans.Resource (MonadResource)
+import Hedgehog (Property, assert)
+import Hedgehog qualified as H
+import Hedgehog.Extras qualified as H
+import Network.JsonRpc.Client.Types ()
+import Spec.Marconi.Sidechain.Utils qualified as U
+import System.Environment qualified as IO
+import System.IO qualified as IO
+import System.Process qualified as IO
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testPropertyNamed)
+
+tests :: TestTree
+tests =
+  testGroup
+    "marconi-sidechain integration test"
+    [ testPropertyNamed
+        "start marconi-sidechain with mandatory cli flags, check all dbs are created and process shuts down correctly"
+        "startMarconiSideChainTest"
+        (startMarconiSideChainTest [] "mandatory-cli-flags")
+    , testPropertyNamed
+        "start marconi-sidechain with all cli flags, check all dbs are created are created and process shuts down correctly"
+        "startMarconiSideChainTest"
+        (startMarconiSideChainTest validOptionalCliFlags "with-optional-cli-flags")
+    ]
+
+validOptionalCliFlags :: [String]
+validOptionalCliFlags =
+  [ "--http-port"
+  , "6660"
+  , "--min-indexing-depth"
+  , "50"
+  , -- "--max-indexing-depth", -- can only have min or max
+    "--addresses-to-index"
+  , "addr1x8phkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gt7r0vd4msrxnuwnccdxlhdjar77j6lg0wypcc9uar5d2shskhj42g"
+  , "--addresses-to-index"
+  , "addr1gx2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer5pnz75xxcrzqf96k"
+  , "--addresses-to-index"
+  , "addr_test1qz2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer3n0d3vllmyqwsx5wktcd8cc3sq835lu7drv2xwl2wywfgs68faae"
+  , "--addresses-to-index"
+  , "addr_test12rphkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gtupnz75xxcryqrvmw"
+  , "--match-asset-id"
+  , "505755e49f9873b02258f1e6fdcc6c81446ab560a0e78180c600d9bd.deadbeef000000000000000000000000000000000000000000000000000000"
+  , "--match-asset-id"
+  , "505755e49f9873b02258f1e6fdcc6c81446ab560a0e78180c600d9bd"
+  , "--fail-if-resyncing-from-genesis"
+  ]
+
+startMarconiSideChainTest :: [String] -> String -> Property
+startMarconiSideChainTest optionalCliArgs description =
+  U.integrationTest $ H.workspace ("start-marconi-sidechain-" <> description) $ \tempPath -> do
```

Comment:
Maybe create a function in Utils such as `integrationTestWithTemp`? Seems like a common pattern. Feel free to ignore if you think it's not useful.

Reply:
As discussed, not useful enough to do. Ignoring.